visualizer/semanticsHelper.js

```diff
@@ -6,26 +6,173 @@
   if (typeof exports === 'object') {
     module.exports = initModule;
   } else {
-    initModule(root.ohmEditor, root.CheckedEmitter);
+    initModule(root.ohm, root.ohmEditor, root.CheckedEmitter);
   }
-})(this, function(ohmEditor, CheckedEmitter) {
-  // Exports
-  // -------
-  ohmEditor.semantics = new CheckedEmitter();
-  ohmEditor.semantics.registerEvents({
-    // Emitted after adding an new operation/attribute
-    'add:semanticOperation': ['type', 'name', 'optArguments']
-  });
+})(this, function(ohm, ohmEditor, CheckedEmitter) {
 
-  // Privates
-  // --------
   var semantics = null;
   ohmEditor.addListener('parse:grammar', function(matchResult, grammar, error) {
     if (grammar && grammar.defaultStartRule) {
       semantics = grammar.semantics();
     }
   });
 
+  var opName = null;
+  var opArguments = null;
+  ohmEditor.semantics.addListener('change:semanticOperation', function(targetName, optArgs) {
+    if (opName === targetName) {
```

Comment:
It's not obvious what this is doing -- why do you set `opName` to null here? Can you maybe add a comment explaining what this is doing?